npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -375,36 +375,22 @@ func (dp *DataPlane) getLocalPodEndpoints() ([]*hcn.HostComputeEndpoint, error)
 	return epPointers, nil
 }
 
-func removeCommonEndpoints(endpoints, endpointsAttached *[]hcn.HostComputeEndpoint) []hcn.HostComputeEndpoint {
-	// Choose smaller and larger lists based on length for efficiency
-	smallerEndpointsList, largerEndpointsList := endpoints, endpointsAttached
-	if len(*endpoints) > len(*endpointsAttached) {
-		smallerEndpointsList, largerEndpointsList = endpointsAttached, endpoints
-	}
-
-	// Store IDs of smaller list in a map for quick lookup
-	idMap := make(map[string]struct{}, len(*smallerEndpointsList))
-	for i := 0; i < len(*smallerEndpointsList); i++ {
-		ep := &(*smallerEndpointsList)[i]
+func GetUniqueEndpoints(endpoints, endpointsAttached []hcn.HostComputeEndpoint) []hcn.HostComputeEndpoint {
+	// Store IDs of endpoints list in a map for quick lookup
+	idMap := make(map[string]struct{}, len(endpoints))
+	for i := 0; i < len(endpoints); i++ {
```

Comment:
not required but golang usually loops through elements with `range` if index isn't needed
```suggestion
	for _, ep := range endpoints {
```

Reply:
changed to use range, thanks

Reply:
@huntergregory, when I updated it to use the range, I would see the following error:
![image](https://github.com/user-attachments/assets/4a93c2dd-e1cc-4e38-8524-dad756ee502c)

I believe it is because the range loop always copies each struct automatically whereas in the for indexed loop, the struct does not get copies automatically. It only gets copies if code is written for it within the for block. 
Having each struct automatically copies results in the error:  rangeValCopy: each iteration copies 392 bytes (consider pointers or indexing) (gocritic)

Reply:
sense

Reply:
That's actually a nice catch from the linter